rust/numaflow-core/src/config.rs

```diff
@@ -42,20 +42,63 @@ pub(crate) struct Settings {
     pub(crate) custom_resource_type: CustomResourceType,
 }
 
+/// LookbackStruct is a struct to hold the lookback window map for the monovertex/pipeline.
+#[derive(Default)]
+pub(crate) struct LookbackStruct {
+    window_map: parking_lot::Mutex<[(&'static str, i64); 4]>,
```

Comment:
why do you need parking_lot?

Reply:
We might be moving to a dynamic approach where this could be updated at runtime. 
Hence, added that for now

Reply:
`LookbackStruct` is built at the startup, it doesn't make sense to make it mutable via mutex. There shouldn't be a reason for the mutable reference to be alive along with a shared reference.